aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -53,16 +53,16 @@ Tensor dequantize_linear_cpu(
   return f;
 }
 
-Scalar q_scale_quant(const Tensor& self) {
+double q_scale_quant(const Tensor& self) {
   auto quantizer = get_qtensorimpl(self)->quantizer();
   TORCH_CHECK(quantizer->qscheme() == kPerTensorAffine);
-  return Scalar(static_cast<PerTensorAffineQuantizer*>(quantizer.get())->scale());
+  return static_cast<PerTensorAffineQuantizer*>(quantizer.get())->scale();
```

Comment:
why are we inconsistent about the types of scale and zero_point?  If you look at Quantizer.h, there are int32_t, int64_t, float, double.  What is our model here?  Does the size actually matter or did we just end up this way because fbgemm takes int32_t, float and native_functions takes int64_t, double?

Reply:
the latter. if we have float and int32_t support in native_functions.yaml I will use that. can we have that?

Reply:
We use double so it's equivalent to the python frontend (which only has doubles).

Given these constraints, this PR doesn't seem worth it.  Either I'd provide a helper or make everything a double.

Reply:
maybe we can cast them into `int32_t` and `float` for fbgemm and keep using `int64_t` and `double`? it is weird to that other aten functions can't use the result of `q_scale()` and `q_zero_point()` directly.

Reply:
You could do that, you should just do validation consistently though.

Reply:
do you mean validation for ranges to zero_point? it's based on scalar type so it should be fine.